net/inadyn/Makefile

```diff
@@ -24,32 +24,40 @@ PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/inadyn
+define Package/inadyn/Default
+  TITLE:=A Dynamic DNS client with SSL/TLS support (uses $(1))
+  VARIANT=$(2)
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+confuse +libopenssl +ca-certificates
-  TITLE:=A Dynamic DNS client with SSL/TLS support
-  URL:=http://troglobit.com/project/inadyn/
   SUBMENU:=IP Addresses and Names
+  DEPENDS:=+confuse +ca-certificates $(3)
+  URL:=https://troglobit.com/projects/inadyn/
 endef
 
-define Package/inadyn/description
-  Inadyn is a small and simple Dynamic DNS, DDNS, client with HTTPS support
-endef
+Package/inadyn-openssl=$(call Package/inadyn/Default,OpenSSL,openssl,+PACKAGE_inadyn-openssl:libopenssl)
+Package/inadyn-mbedtls=$(call Package/inadyn/Default,Mbed TLS,mbedtls,+PACKAGE_inadyn-mbedtls:libmbedtls)
 
-define Package/inadyn/conffiles
-/etc/inadyn.conf
-endef
+CONFIGURE_ARGS += \
+  --$(if $(CONFIG_PACKAGE_inadyn-openssl),en,dis)able-openssl \
+  --$(if $(CONFIG_PACKAGE_inadyn-mbedtls),en,dis)able-mbedtls \
+  --enable-shared \
+  --disable-static \
+  --with-pic
 
+ifneq ($(CONFIG_PACKAGE_inadyn-mbedtls),)
 CONFIGURE_ARGS += \
-	--enable-shared \
-	--disable-static \
-	--enable-openssl \
-	--with-pic
+  MbedTLS_CFLAGS="-I$(STAGING_DIR)/usr/include" \
```

Comment:
curious about this. why?

Reply:
inadyn's configure script fails to automatically find mbedtls because it looks like mbedtls doesn't provide a pkg-config file.

Is there some better way than manually specifying these compile flags?

Reply:
modify the original configure script to look for libraries instead of using pkg-config.